src/Analyzer.Utilities/Extensions/ISymbolExtensions.cs

```diff
@@ -147,6 +150,118 @@ public static bool MatchFieldByName(this ISymbol member, INamedTypeSymbol type,
         {
             return member != null && member.Kind == SymbolKind.Field && member.MatchMemberByName(type, name);
         }
+
+        public static bool ContainsParameterWithType(this IEnumerable<IParameterSymbol> parameters, INamedTypeSymbol type)
+        {
+            var parametersWithType = GetParametersWithType(parameters, type);
+            return parametersWithType.Any();
+        }
+
+        public static IEnumerable<IParameterSymbol> GetParametersWithType(this IEnumerable<IParameterSymbol> parameters, INamedTypeSymbol type)
+        {
+            return parameters.Where(p => p.Type?.Equals(type) == true);
+        }
+
+        public static bool OverloadWithGivenTypeParameterExist(this IEnumerable<IMethodSymbol> overloads, IMethodSymbol self, INamedTypeSymbol type, CancellationToken cancellationToken)
```

Comment:
I got confused by the name - suggests that it is looking for an overload given type parameter, while the implementation is actually looking for any parameter with a given type.